### PR TITLE
Jupyter checkpoint

### PR DESCRIPTION
**Pull request by PrajwalPrashanth**
* Explicitly saving the checkpoint from extension

TODO: 
* Some times even the `jovian.commit` fails to capture the latest checkpoint. Have to look about that

### CODE REVIEW

**Comment by aakashns** [comment]
In `jovian.commit`, we invoke a Javascript function. It fails for password protected notebooks. Can you try checking if we can look at the output of the javascript and detect whether it fails? 

If it does fail, let's show a warning to the user asking them to save the notebook. Try and see if we can do this.

File: jovian/jovian_nb_ext/main.js [hunk 1]
```diff
@@ -28,6 +28,8 @@ define([
         const jvnLog = data => {
           resolve(data.content.text.trim());
         };
+        /* Saves checkpoints before committing*/
+        Jupyter.notebook.save_checkpoint();
```
Comment (aakashns) [inline jovian/jovian_nb_ext/main.js]: Better give a 1-2 second delay before continuing, because this is an asynchronous action.

Comment (aakashns) [reply jovian/jovian_nb_ext/main.js]: @PrajwalPrashanth did you get a chance to look into this?

**Comment by aakashns** [comment]
@PrajwalPrashanth  did you get a chance to look into this?

File: jovian/jovian_nb_ext/main.js (hunk 1 above)
Comment (sidujjain) [reply jovian/jovian_nb_ext/main.js]: can you see if you wait for this event: https://github.com/jupyter/notebook/blob/8b21329deb9dd04271b12e3d2790c5be9f1fd51e/notebook/static/notebook/js/notebook.js#L2867

Comment (PrajwalPrashanth) [reply jovian/jovian_nb_ext/main.js]: image

Yes we can do that before committing, shall we do it for the library as it won't require explicitly doing this in the extension.

Comment (PrajwalPrashanth) [reply jovian/jovian_nb_ext/main.js]: But if i commit for that event then it commit for every autosave right?

Comment (sidujjain) [reply jovian/jovian_nb_ext/main.js]: we can do it for extension for now, can do for library in a separate PR

Comment (PrajwalPrashanth) [reply jovian/jovian_nb_ext/main.js]: Done.

**Comment by PrajwalPrashanth** [comment]
`pip install git+https://github.com/JovianML/jovian-py.git@pp/ch708/jupyter-checkpoint-commit --upgrade` 
installation code to test it out

**Pull request merged**